src/js/main.js

```diff
@@ -0,0 +1,53 @@
+import auth from './auth';
+
+const loginPopup = document.getElementById('login-popup');
+const logoutButton = document.getElementById('logout-btn');
+const errorContainer = document.getElementById('email-error');
+const loginForm = document.getElementById('login-form');
+
+const isValidEmail = (email) => /^[^\s@]+@[^\s@]+\.[^\s@]+$/.test(email);
+
+const hideLoginPopup = () => {
+  loginPopup.style.display = 'none';
+};
+
+const displayLoginPopup = () => {
+  loginPopup.style.display = 'flex';
+};
+
+const handleAuthentication = () => {
+  const session = auth.getSession();
+
+  if (session) {
+    hideLoginPopup();
+  } else {
+    displayLoginPopup();
+  }
+};
+
+const handleLogout = () => {
+  auth.signOut();
+  handleAuthentication();
+};
+
+const handleSubmit = (event) => {
+  event.preventDefault();
+
+  const email = document.getElementById('email').value.trim();
+
+  if (isValidEmail(email)) {
+    auth.signIn(email);
+    hideLoginPopup();
```

Comment:
you should attempt to sign in first, if the value is not value signin will return error then you should now decide if you will hide the popup or show an error message.